### ISSUE TITLE
Updating assignments gives false-positive

### ISSUE THREAD

**Issue by osmoc** [koalaman/shellcheck#944]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1068
- My shellcheck version (`shellcheck --version` or "online"): online
- [ x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh
$conf_flags += $*

```

#### Here's what shellcheck currently says:
conf_flags="--with-feature"
$conf_flags += $*
    ^-- SC2039: In POSIX sh, += is undefined.
     ^-- SC1066: Don't use $ on the left side of assignments.
                ^-- SC1068: Don't put spaces around the = in assignments.

#### Here's what I wanted or expected to see:
I don't think SC1066 applies in this case, not sure about SC1068.

**Comment by brother** [comment]
```
brother ~$ /tmp/apa.sh 1 2 3
/tmp/apa.sh: 2: /tmp/apa.sh: +=1: not found

brother ~$ cat /tmp/apa.sh 
#!/bin/sh
$apa+=$*

echo $apa
```

To me the warning looks correct.
```
brother ~$ cat /tmp/apa.sh 
#!/bin/sh
$apa += $*

echo $apa
brother ~$ /tmp/apa.sh 
/tmp/apa.sh: 2: /tmp/apa.sh: +=: not found
```

**Comment by koalaman** [comment]
It should definitely be saying "+=" instead of "=", but other than that it's all valid

**Comment by osmoc** [comment]
Isn't it bashism and should be reported as such?

**Comment by koalaman** [comment]
Yes. The first error was about that:

    SC2039: In POSIX sh, += is undefined
